### PR TITLE
Unify interface keyword args of metadata commands according to issue #6461

### PR DESCRIPTION
**Pull request by christian-monch**
This PR contains keyword unification work (issue #6461) of the metadata modules

#### 🏠 Internal
- ensure that internal datalad command calls do write results to stdout

### CODE REVIEW

File: datalad/metadata/search.py [hunk 1]
```diff
@@ -286,7 +286,12 @@ def _search_from_virgin_install(dataset, query):
                      "superdataset at %r?"
                      % DEFAULT_DATASET_PATH):
             from datalad.api import install
-            default_ds = install(DEFAULT_DATASET_PATH, source='///')
+            default_ds = install(
+                DEFAULT_DATASET_PATH,
+                source='///',
+                result_renderer='disabled',
+                on_failur='continue',
```
Comment (yarikoptic) [inline datalad/metadata/search.py:293]:
```suggestion
                on_failure='continue',
```

**Comment by mih** [comment]
```
======================================================================
2629ERROR: datalad.metadata.tests.test_base.test_aggregation
2630----------------------------------------------------------------------
2631Traceback (most recent call last):
2632  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/nose/case.py", line 198, in runTest
2633    self.test(*self.arg)
2634  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/tests/utils.py", line 579, in _wrap_with_tree
2635    return t(*(arg + (d,)), **kw)
2636  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/metadata/tests/test_base.py", line 177, in test_aggregation
2637    cloneres = clone.metadata()
2638  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/distribution/dataset.py", line 502, in apply_func
2639    return f(*args, **kwargs)
2640  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/interface/utils.py", line 447, in eval_func
2641    return return_func(*args, **kwargs)
2642  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/interface/utils.py", line 439, in return_func
2643    results = list(results)
2644  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/interface/utils.py", line 357, in generator_func
2645    for r in _process_results(
2646  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/interface/utils.py", line 544, in _process_results
2647    for res in results:
2648  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/metadata/metadata.py", line 991, in __call__
2649    for r in query_aggregated_metadata(
2650  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/metadata/metadata.py", line 309, in query_aggregated_metadata
2651    dsmeta = _load_json_object(
2652  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/metadata/metadata.py", line 115, in _load_json_object
2653    jsonload(fpath, fixup=True) if op.lexists(fpath) else {})
2654  File "/tmp/dl-miniconda-dlfjv27j/lib/python3.9/site-packages/datalad/support/json_py.py", line 183, in load
2655    with _suitable_open(fname, compressed)(fname, 'rb') as f:
2656FileNotFoundError: [Errno 2] No such file or directory: '/tmp/datalad_temp_tree_test_aggregationtm5xzgii/clone/.datalad/metadata/objects/cd/ds-c19d0ef6aeba7a5f2258c05bc8b01c'
```

shows up consistently across CIs, probably real.

File: datalad/metadata/search.py [hunk 2]
```diff
@@ -286,7 +286,12 @@ def _search_from_virgin_install(dataset, query):
                      "superdataset at %r?"
                      % DEFAULT_DATASET_PATH):
             from datalad.api import install
-            default_ds = install(DEFAULT_DATASET_PATH, source='///')
+            default_ds = install(
```
Comment (yarikoptic) [inline datalad/metadata/search.py:289]: Return type is generator now, so it should no longer return a dataset here, but tests are green. What am I missing?

**Comment by christian-monch** [comment]
> 
> shows up consistently across CIs, probably real.

Thx, fixed

File: datalad/metadata/search.py (hunk 2 above)
Comment (christian-monch) [reply datalad/metadata/search.py:289]: There is only one test that touches this path and in this test `install` is mocked out. I will have a close look at this

Converting it to draft

Comment (christian-monch) [reply datalad/metadata/search.py:289]: Due to the expected behavior, a `generator` result-type is not usefu here.

**Comment by christian-monch** [comment]
> ```
> ======================================================================
> 7247FAIL: datalad.metadata.tests.test_search.test_search_outside1_install_default_ds
> 7248----------------------------------------------------------------------
> 7249Traceback (most recent call last):
> 7250  File "/home/appveyor/dlvenv/lib/python3.8/site-packages/nose/case.py", line 198, in runTest
> 7251    self.test(*self.arg)
> 7252  File "/home/appveyor/projects/datalad/datalad/tests/utils.py", line 1611, in _wrap_with_testsui
> 7253    ret = t(*args, **kwargs)
> 7254  File "/home/appveyor/projects/datalad/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
> 7255    return t(*(arg + (filename,)), **kw)
> 7256  File "/home/appveyor/projects/datalad/datalad/tests/utils.py", line 874, in _wrap_with_tempfile
> 7257    return t(*(arg + (filename,)), **kw)
> 7258  File "/home/appveyor/projects/datalad/datalad/metadata/tests/test_search.py", line 97, in test_search_outside1_install_default_ds
> 7259    _check_mocked_install(default_dspath, mock_install)
> 7260  File "/home/appveyor/projects/datalad/datalad/metadata/tests/test_search.py", line 153, in _check_mocked_install
> 7261    mock_install.assert_called_once_with(
> 7262  File "/usr/lib/python3.8/unittest/mock.py", line 925, in assert_called_once_with
> 7263    return self.assert_called_with(*args, **kwargs)
> 7264  File "/usr/lib/python3.8/unittest/mock.py", line 913, in assert_called_with
> 7265    raise AssertionError(_error_message()) from cause
> 7266AssertionError: expected call not found.
> 7267Expected: install('/home/appveyor/DLTMP/datalad_temp_test_search_outside1_install_default_ds0q6uyy4s', source='///', result_renderer='disabled', on_failure='continue', result_type='generator')
> 7268Actual: install('/home/appveyor/DLTMP/datalad_temp_test_search_outside1_install_default_ds0q6uyy4s', source='///', result_renderer='disabled')
> ```
> 
> One more expectation mismatch

Weird, it is not in the code on my side. Will force-push and check back

**Comment by christian-monch** [comment]
Mystery solved: I checked the wrong local branch (too many issue-6461-... branches here) :-/

**Comment by christian-monch** [comment]
Only errors left are on MacOS and seem to be related to git-annex.

**Review by mih — APPROVED** [review]
LGTM, thx!

File: datalad/metadata/search.py (hunk 2 above)
Comment (christian-monch) [reply datalad/metadata/search.py:289]: finally changed to generator return type

**Pull request merged**